### PR TITLE
Fix remaining warnings for Qt 5.15

### PR DESCRIPTION
**Pull request by Holzhaus**
With these changes, Mixxx is now warning-free again (except for the restrict warnings in the effect code).

### CODE REVIEW

File: src/network/jsonwebtask.cpp [hunk 1]
```diff
@@ -274,10 +274,10 @@ bool JsonWebTask::doStart(
             Qt::UniqueConnection);
 
     connect(m_pendingNetworkReply,
-#if QT_VERSION >= QT_VERSION_CHECK(5, 7, 0)
-            qOverload<QNetworkReply::NetworkError>(&QNetworkReply::error),
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
+            &QNetworkReply::errorOccurred,
 #else
-            QOverload<QNetworkReply::NetworkError>::of(&QNetworkReply::error),
+            qOverload<QNetworkReply::NetworkError>::of(&QNetworkReply::error),
```
Comment (daschuer) [inline src/network/jsonwebtask.cpp:280]: Typo: Capital Q

File: src/musicbrainz/web/musicbrainzrecordingstask.cpp [hunk 2]
```diff
@@ -124,7 +124,11 @@ bool MusicBrainzRecordingsTask::doStart(
             Qt::UniqueConnection);
 
     connect(m_pendingNetworkReply,
-            QOverload<QNetworkReply::NetworkError>::of(&QNetworkReply::error),
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
+            &QNetworkReply::errorOccurred,
+#else
+            qOverload<QNetworkReply::NetworkError>::of(&QNetworkReply::error),
```
Comment (daschuer) [inline src/musicbrainz/web/musicbrainzrecordingstask.cpp:130]: Typo: Capital Q

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you for taking care.

File: src/network/jsonwebtask.cpp (hunk 1 above)
Comment (Holzhaus) [reply src/network/jsonwebtask.cpp:280]: Ouch, I wonder why it compiled locally...

Comment (Holzhaus) [reply src/network/jsonwebtask.cpp:280]: Because I already have Qt 5.15... Nevermind

**Comment by Holzhaus** [comment]
I resolved the review comments. Please recheck as I didn't test these changes (Currently waiting in front of a bike repair shop and I don't think Mixxx would build on my phone...at least vim and git works :sunglasses:).

**Review by uklotzde — APPROVED** [review]

**Review by daschuer — APPROVED** [review]
LGTM, thank you for doing this piece of maintenance.

**Pull request merged**